### ISSUE TITLE
Audio and Video suggestion

### ISSUE THREAD

**Issue by gedw99** [fyne-io/fyne#612]
I just watrched the youtiube presenation series of fyne.

Wow !!! 

For the 2.0 release i think that pion webrtc is the perfect marriage. 
I bet the Webrtc pion tram would love to team up on this.

**Comment by gedw99** [comment]
https://github.com/pion/

**Comment by gedw99** [comment]
actually i just realised the there are only two realistic ways to do it.
1. use the chromoium webrtc dll
2. Use the zserge Webview
otherwise its going to be boiling the ocean because google spent so many years to get for example audio fitter perfected

Option 2 is much easier and they just got it working for Edge Webview for Windows 10.
https://github.com/zserge/webview/tree/webview-x

**Comment by gedw99** [comment]
Might also need this but i doubt it if you use the zserge approach, because the webview will get you access to MIC and WebCam: https://github.com/pion/mediadevices

**Comment by andydotxyz** [comment]
I just read more on this and I think I did not understand before.
Is this about audio and video decoding and playing for all things, or is it specific for web UI? There are a lot of things talking about webview etc so I was not sure.

**Comment by Jacalz** [comment]
Yeah. Webview and bundling chromium are entirely out of the question.

**Comment by gedw99** [comment]
> I just read more on this and I think I did not understand before. Is this about audio and video decoding and playing for all things, or is it specific for web UI? There are a lot of things talking about webview etc so I was not sure.

 https://github.com/pion/mediadevices is designed to NOT need a web browser or Web View.

The recent work in GIo for Video and audio  playing ( again no need for web ) opens the door for GIO and FYNE apps to look at making WebRTC video conf, but also just video and audio playing.
https://github.com/gioui/gio-example/pull/2

webrtc Sound jitter management is harder.

**Comment by andydotxyz** [comment]
You might be interested in this demo that I put together https://github.com/fyshos/movies.
I think it uses the same underlying library as it was a Gio comment that pointed it out :).
I've not noticed any audio or video jitter so far (though I did push some build fixes upstream).

**Comment by gedw99** [comment]
nice one @andydotxyz 

Will have to give it a try.

also using github.com/zergon321/reisen

Its fantastic how Fyne and GIO are able to leverage the same stuff. Just like the Text shaping / halfbuz work.

**Comment by andydotxyz** [comment]
> Its fantastic how Fyne and GIO are able to leverage the same stuff. Just like the Text shaping / halfbuz work.

Yes, I think it is a testament to the great Go community and the quality and collaboration on excellent libraries.
